tests/wpt/web-platform-tests/html/semantics/embedded-content/media-elements/event_pause_noautoplay.html

```diff
@@ -18,21 +18,25 @@
   var t = async_test("calling play() then pause() on non-autoplay audio should trigger pause event", {timeout:5000});
   var a = document.getElementById("a");
   a.addEventListener("pause", function() {
-    t.done();
+    play_promise.then(function () {
+      assert_unreached("pause() will reject all pending play promises.");
+    }, t.step_func_done());
   }, false);
   a.src = getAudioURI("/media/sound_5") + "?" + new Date() + Math.random();
-  a.play();
+  var play_promise = a.play();
```

Comment:
Let's not move the `Promise` to a `var` and handle this inline:

```javascript
a.play()
.then(t.step_func(function() {
  assert_unreached("pause() should reject all pending play Promises");
}))
.catch(t.step_func(function() {
  assert_true(true, "pause() should reject all pending play Promises");
}));
```

Reply:
This comment applies to all the other tests with a similar pattern :)

Reply:
Good solution! Let me update all of them :)

Btw, I think the `catch` might need to be like following?

```js
.catch(t.step_func_done(function() {
  assert_true(true, "pause() should reject all pending play Promises");
}));
```